test/jpypetest/test_jvmfinder.py

```diff
@@ -66,34 +64,113 @@ def test_get_from_bin(self, mock_real_path, mock_path_exists, mock_os_walk):
         mock_path_exists.return_value = True
         mock_real_path.return_value = '/usr/lib/jvm/java-6-openjdk-amd64/bin/java'
 
-        finder = LinuxJVMFinder()
+        finder = jpype._jvmfinder.LinuxJVMFinder()
         p = finder._get_from_bin()
 
         self.assertEqual(
             p, os.path.join('/usr/lib/jvm/java-6-openjdk-amd64/jre/lib/amd64/server', 'libjvm.so'))
 
     @mock.patch('platform.mac_ver')
-    def test_javahome_binary_py27(self, mock_mac_ver):
+    def testDarwinBinary(self, mock_mac_ver):
         # this version has java_home binary
         mock_mac_ver.return_value = ('10.6.8', '', '')
 
         expected = '/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home\n'
 
-        finder = DarwinJVMFinder()
+        finder = jpype._jvmfinder.DarwinJVMFinder()
 
         # fake check_output
         with mock.patch('subprocess.check_output') as mock_checkoutput:
             mock_checkoutput.return_value = expected
             p = finder._javahome_binary()
-
             self.assertEqual(p, expected.strip())
 
         # this version has no java_home binary
         mock_mac_ver.return_value = ('10.5', '', '')
         p = finder._javahome_binary()
-
         self.assertEqual(p, None)
 
+    @common.unittest.skip
+    def testPlatform(self):
```

Comment:
This tests a implementation detail.

Reply:
This is mostly just to make sure that coverage tool hits all code.  I really don't care about the result other that making sure that we don't have a syntax error down one of the paths (which in one case there was a typo that only hit on one platform.)

Would you like me to put a decorator on all those tests to be "coverageOnly"  so that they only trigger during the coverage phase?

Reply:
If it is just in place to test coverage/ e.g. correct syntax I would just ensure that the code paths are all hit. This code path is only used on windows, right? So it would be sufficient to test coverage on Appveyor as well. Codecov does combine reports from different platforms/configurations without objection.

Reply:
Not sure I trust appveyor to run a code coverage.   It is a tradeoff.  Either I make a somewhat useless test on Linux (mocking to see if the paths get hit) with the other tests for coverage.  Or I face the challenge of getting code coverage to work on windows.   I think that given the hours we tend to burn on making the CI work (push, wait 15 minutes, see it didn't work, push, rinse, lather, repeat) making it work once on one system is likely the best option.  We have both put in a huge set of commit after commit pushing one line in over and over trying to get the CI working, so I favor the somewhat useless mock test approach.

Reply:
I agree that setting up CI can be tedious work. But to simply check for a well running function, you do not need to check the details, right?
M$ VC can also produce coverage (binary files), which can be converted to xml by another tool (yay). https://github.com/jsargiot/visual-coverage
But please do not waste your precious time on this BS :)

Reply:
Sorry I got you wrong, you have to mock to actually trigger this code paths. I just didn't like the struct packing binary thing as it seemed redundant with the actual implementation.

Reply:
Oh the mock packing.   It was just a way to try to fake a binary file that I don't have.  The images on the CI don't have the wrong architecture loaded (32bit on 64bit, etc)  Thus I either put those images into the test bench (dubious licence wise) or use mock to fake the bytes in the right place.  I was planning to the same thing for elf as windows pei files so that we cover it.    I suppose I could make short header file only versions of the libraries for each architecture.) 

I can make it go through the entry point rather than hitting the internal function directly if you want.   I just have to mock it to windows and call the top level function.  That one is no problem.

(I am guess you want to refer to testCheckArch)

Reply:
Promoting this to issue.